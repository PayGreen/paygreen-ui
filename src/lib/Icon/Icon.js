import React from 'react';
import IconWrapper from './IconWrapper';
import {
  ArrowBottom,
  ArrowCircleLeft,
  ArrowCircleLeftBold,
  ArrowCircleRightBold,
  ArrowCircleRight,
  ArrowLeft,
  ArrowLongBottom,
  ArrowLongLeft,
  ArrowLongRight,
  ArrowLongTop,
  ArrowRight,
  ArrowRoundBottomLeft,
  ArrowRoundBottomRight,
  ArrowRoundTopLeft,
  ArrowRoundTopRight,
  ArrowTop,
  BalloonBold,
  Balloon,
  Bank,
  BeakerBold,
  Beaker,
  BellBold,
  Bell,
  BulbBold,
  Bulb,
  CardsBold,
  Cards,
  Car,
  Cart,
  CautionBold,
  Caution,
  Change,
  CheckBold,
  Checkbox,
  Check,
  ClockBold,
  Clock,
  Code,
  Coins,
  Contract,
  CrossBold,
  Cross,
  Download,
  Empty,
  Eye,
  Factory,
  Globe,
  Heart,
  Hourglass,
  Information,
  Leaf,
  LetterBold,
  Letter,
  LinkBold,
  Link,
  List,
  Lock,
  Marker,
  Meal,
  Megaphone,
  MenuBold,
  MenuDots,
  Menu,
  Meter,
  Minus,
  Moon,
  Moove,
  Organization,
  Out,
  Paperclip,
  Pen,
  Phone,
  Picture,
  Plane,
  Plus,
  Pointer,
  Popin,
  QuestionBold,
  Question,
  QuoteLeft,
  QuoteRight,
  RefreshBold,
  Refresh,
  Satchel,
  Schedule,
  SearchBold,
  Search,
  Shop,
  Sun,
  Support,
  Trees,
  Trophy,
  Truck,
  Tv,
  UserBold,
  Users,
  User,
  Wallet,
  Weight,
  Wheels,
} from './components';

export const ArrowBottomIcon = (props) => <IconWrapper {...props}><ArrowBottom /></IconWrapper>
export const ArrowCircleLeftIcon = (props) => <IconWrapper {...props}><ArrowCircleLeft /></IconWrapper>
export const ArrowCircleLeftBoldIcon = (props) => <IconWrapper {...props}><ArrowCircleLeftBold /></IconWrapper>
export const ArrowCircleRightBoldIcon = (props) => <IconWrapper {...props}><ArrowCircleRightBold /></IconWrapper>
export const ArrowCircleRightIcon = (props) => <IconWrapper {...props}><ArrowCircleRight /></IconWrapper>
export const ArrowLeftIcon = (props) => <IconWrapper {...props}><ArrowLeft /></IconWrapper>
export const ArrowLongBottomIcon = (props) => <IconWrapper {...props}><ArrowLongBottom /></IconWrapper>
export const ArrowLongLeftIcon = (props) => <IconWrapper {...props}><ArrowLongLeft /></IconWrapper>
export const ArrowLongRightIcon = (props) => <IconWrapper {...props}><ArrowLongRight /></IconWrapper>
export const ArrowLongTopIcon = (props) => <IconWrapper {...props}><ArrowLongTop /></IconWrapper>
export const ArrowRightIcon = (props) => <IconWrapper {...props}><ArrowRight /></IconWrapper>
export const ArrowRoundBottomLeftIcon = (props) => <IconWrapper {...props}><ArrowRoundBottomLeft /></IconWrapper>
export const ArrowRoundBottomRightIcon = (props) => <IconWrapper {...props}><ArrowRoundBottomRight /></IconWrapper>
export const ArrowRoundTopLeftIcon = (props) => <IconWrapper {...props}><ArrowRoundTopLeft /></IconWrapper>
export const ArrowRoundTopRightIcon = (props) => <IconWrapper {...props}><ArrowRoundTopRight /></IconWrapper>
export const ArrowTopIcon = (props) => <IconWrapper {...props}><ArrowTop /></IconWrapper>
export const BalloonBoldIcon = (props) => <IconWrapper {...props}><BalloonBold /></IconWrapper>
export const BalloonIcon = (props) => <IconWrapper {...props}><Balloon /></IconWrapper>
export const BankIcon = (props) => <IconWrapper {...props}><Bank /></IconWrapper>
export const BeakerBoldIcon = (props) => <IconWrapper {...props}><BeakerBold /></IconWrapper>
export const BeakerIcon = (props) => <IconWrapper {...props}><Beaker /></IconWrapper>
export const BellBoldIcon = (props) => <IconWrapper {...props}><BellBold /></IconWrapper>
export const BellIcon = (props) => <IconWrapper {...props}><Bell /></IconWrapper>
export const BulbBoldIcon = (props) => <IconWrapper {...props}><BulbBold /></IconWrapper>
export const BulbIcon = (props) => <IconWrapper {...props}><Bulb /></IconWrapper>
export const CardsBoldIcon = (props) => <IconWrapper {...props}><CardsBold /></IconWrapper>
export const CardsIcon = (props) => <IconWrapper {...props}><Cards /></IconWrapper>
export const CarIcon = (props) => <IconWrapper {...props}><Car /></IconWrapper>
export const CartIcon = (props) => <IconWrapper {...props}><Cart /></IconWrapper>
export const CautionBoldIcon = (props) => <IconWrapper {...props}><CautionBold /></IconWrapper>
export const CautionIcon = (props) => <IconWrapper {...props}><Caution /></IconWrapper>
export const ChangeIcon = (props) => <IconWrapper {...props}><Change /></IconWrapper>
export const CheckBoldIcon = (props) => <IconWrapper {...props}><CheckBold /></IconWrapper>
export const CheckboxIcon = (props) => <IconWrapper {...props}><Checkbox /></IconWrapper>
export const CheckIcon = (props) => <IconWrapper {...props}><Check /></IconWrapper>
export const ClockBoldIcon = (props) => <IconWrapper {...props}><ClockBold /></IconWrapper>
export const ClockIcon = (props) => <IconWrapper {...props}><Clock /></IconWrapper>
export const CodeIcon = (props) => <IconWrapper {...props}><Code /></IconWrapper>
export const CoinsIcon = (props) => <IconWrapper {...props}><Coins /></IconWrapper>
export const ContractIcon = (props) => <IconWrapper {...props}><Contract /></IconWrapper>
export const CrossBoldIcon = (props) => <IconWrapper {...props}><CrossBold /></IconWrapper>
export const CrossIcon = (props) => <IconWrapper {...props}><Cross /></IconWrapper>
export const DownloadIcon = (props) => <IconWrapper {...props}><Download /></IconWrapper>
export const EmptyIcon = (props) => <IconWrapper {...props}><Empty /></IconWrapper>
export const EyeIcon = (props) => <IconWrapper {...props}><Eye /></IconWrapper>
export const FactoryIcon = (props) => <IconWrapper {...props}><Factory /></IconWrapper>
export const GlobeIcon = (props) => <IconWrapper {...props}><Globe /></IconWrapper>
export const HeartIcon = (props) => <IconWrapper {...props}><Heart /></IconWrapper>
export const HourglassIcon = (props) => <IconWrapper {...props}><Hourglass /></IconWrapper>
export const InformationIcon = (props) => <IconWrapper {...props}><Information /></IconWrapper>
export const LeafIcon = (props) => <IconWrapper {...props}><Leaf /></IconWrapper>
export const LetterBoldIcon = (props) => <IconWrapper {...props}><LetterBold /></IconWrapper>
export const LetterIcon = (props) => <IconWrapper {...props}><Letter /></IconWrapper>
export const LinkBoldIcon = (props) => <IconWrapper {...props}><LinkBold /></IconWrapper>
export const LinkIcon = (props) => <IconWrapper {...props}><Link /></IconWrapper>
export const ListIcon = (props) => <IconWrapper {...props}><List /></IconWrapper>
export const LockIcon = (props) => <IconWrapper {...props}><Lock /></IconWrapper>
export const MarkerIcon = (props) => <IconWrapper {...props}><Marker /></IconWrapper>
export const MealIcon = (props) => <IconWrapper {...props}><Meal /></IconWrapper>
export const MegaphoneIcon = (props) => <IconWrapper {...props}><Megaphone /></IconWrapper>
export const MenuBoldIcon = (props) => <IconWrapper {...props}><MenuBold /></IconWrapper>
export const MenuDotsIcon = (props) => <IconWrapper {...props}><MenuDots /></IconWrapper>
export const MenuIcon = (props) => <IconWrapper {...props}><Menu /></IconWrapper>
export const MeterIcon = (props) => <IconWrapper {...props}><Meter /></IconWrapper>
export const MinusIcon = (props) => <IconWrapper {...props}><Minus /></IconWrapper>
export const MoonIcon = (props) => <IconWrapper {...props}><Moon /></IconWrapper>
export const MooveIcon = (props) => <IconWrapper {...props}><Moove /></IconWrapper>
export const OrganizationIcon = (props) => <IconWrapper {...props}><Organization /></IconWrapper>
export const OutIcon = (props) => <IconWrapper {...props}><Out /></IconWrapper>
export const PaperclipIcon = (props) => <IconWrapper {...props}><Paperclip /></IconWrapper>
export const PenIcon = (props) => <IconWrapper {...props}><Pen /></IconWrapper>
export const PhoneIcon = (props) => <IconWrapper {...props}><Phone /></IconWrapper>
export const PictureIcon = (props) => <IconWrapper {...props}><Picture /></IconWrapper>
export const PlaneIcon = (props) => <IconWrapper {...props}><Plane /></IconWrapper>
export const PlusIcon = (props) => <IconWrapper {...props}><Plus /></IconWrapper>
export const PointerIcon = (props) => <IconWrapper {...props}><Pointer /></IconWrapper>
export const PopinIcon = (props) => <IconWrapper {...props}><Popin /></IconWrapper>
export const QuestionBoldIcon = (props) => <IconWrapper {...props}><QuestionBold /></IconWrapper>
export const QuestionIcon = (props) => <IconWrapper {...props}><Question /></IconWrapper>
export const QuoteLeftIcon = (props) => <IconWrapper {...props}><QuoteLeft /></IconWrapper>
export const QuoteRightIcon = (props) => <IconWrapper {...props}><QuoteRight /></IconWrapper>
export const RefreshBoldIcon = (props) => <IconWrapper {...props}><RefreshBold /></IconWrapper>
export const RefreshIcon = (props) => <IconWrapper {...props}><Refresh /></IconWrapper>
export const SatchelIcon = (props) => <IconWrapper {...props}><Satchel /></IconWrapper>
export const ScheduleIcon = (props) => <IconWrapper {...props}><Schedule /></IconWrapper>
export const SearchBoldIcon = (props) => <IconWrapper {...props}><SearchBold /></IconWrapper>
export const SearchIcon = (props) => <IconWrapper {...props}><Search /></IconWrapper>
export const ShopIcon = (props) => <IconWrapper {...props}><Shop /></IconWrapper>
export const SunIcon = (props) => <IconWrapper {...props}><Sun /></IconWrapper>
export const SupportIcon = (props) => <IconWrapper {...props}><Support /></IconWrapper>
export const TreesIcon = (props) => <IconWrapper {...props}><Trees /></IconWrapper>
export const TrophyIcon = (props) => <IconWrapper {...props}><Trophy /></IconWrapper>
export const TruckIcon = (props) => <IconWrapper {...props}><Truck /></IconWrapper>
export const TvIcon = (props) => <IconWrapper {...props}><Tv /></IconWrapper>
export const UserBoldIcon = (props) => <IconWrapper {...props}><UserBold /></IconWrapper>
export const UsersIcon = (props) => <IconWrapper {...props}><Users /></IconWrapper>
export const UserIcon = (props) => <IconWrapper {...props}><User /></IconWrapper>
export const WalletIcon = (props) => <IconWrapper {...props}><Wallet /></IconWrapper>
export const WeightIcon = (props) => <IconWrapper {...props}><Weight /></IconWrapper>
export const WheelsIcon = (props) => <IconWrapper {...props}><Wheels /></IconWrapper>
