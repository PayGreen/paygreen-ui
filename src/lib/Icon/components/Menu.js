import React from "react";

const SvgMenu = props => (
  <svg viewBox="0 0 143 143" {...props}>
    <path d="M114.8 45.6H26.2c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5h88.7c1.9 0 3.5 1.6 3.5 3.5s-1.6 3.5-3.6 3.5zM114.8 76.9H26.2c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5h88.7c1.9 0 3.5 1.6 3.5 3.5s-1.6 3.5-3.6 3.5zM114.8 107.9H26.2c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5h88.7c1.9 0 3.5 1.6 3.5 3.5s-1.6 3.5-3.6 3.5z" />
  </svg>
);

export default SvgMenu;
