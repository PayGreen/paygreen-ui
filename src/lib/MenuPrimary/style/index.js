import styled from 'styled-components';
import { menuBlock, closedStyle } from './base';
import { MenuBase } from '../../Menu/style';

const MenuPrimaryBase = styled.ul`
    ${menuBlock};

    @media (${props => props.theme.query.max.lg}) {
        height: calc(100vh - ${props => props.theme.grid.header});
        padding: ${props => props.theme.space.sm} 0;
        
        ${props => props.isOpen ? null : closedStyle};
    }

    @media (${props => props.theme.query.min.lg}) {
        ${MenuBase} {
            &:hover {
                /* Let MenuList open when cursor is between MenuItem and MenuList */
                &:after {
                    content: '';
                    position: absolute;
                    width: 100%;
                    height: ${props => props.theme.iconSize.lg};
                    bottom: 0;
                    transform: translateY(100%);
                    z-index: ${props => props.theme.zindex.menu};
                }
            }
        }
    }
`;

export { MenuPrimaryBase };