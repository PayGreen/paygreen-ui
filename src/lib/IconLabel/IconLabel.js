import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import {
    colorThemeOptions,
    colorThemeDefault,
    spaceOptions,
    spaceDefault,
} from '../../shared/constants';
import { IconLabelBase } from './style';

class IconLabel extends PureComponent {
    render() {
        return <IconLabelBase
            {...this.props}
        >
            {this.props.children}
        </IconLabelBase>;
    }
}

IconLabel.propTypes = {
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
    marginLeft: PropTypes.oneOf(Object.values(spaceOptions)),
    marginRight: PropTypes.oneOf(Object.values(spaceOptions)),
};

IconLabel.defaultProps = {
    colorTheme: colorThemeDefault,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,
    marginLeft: spaceDefault,
    marginRight: spaceDefault,
};

export default IconLabel;