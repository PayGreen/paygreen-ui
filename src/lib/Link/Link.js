import React from 'react';
import PropTypes from 'prop-types';
import {
    colorTypeOptions,
    colorTypeDefault,
    colorPalletOptions,
    colorPalletDefault,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    greyDefault,
    formStatusOptions,
    formStatusDefault,
} from '../../shared/constants';
import { LinkBase } from './style';

const Link = props => {
    return <LinkBase {...props}>{props.children}</LinkBase>;
};

Link.propTypes = {
    colorType: PropTypes.oneOf(Object.values(colorTypeOptions)),
    colorPallet: PropTypes.oneOf(Object.values(colorPalletOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    colorWab: PropTypes.oneOf(Object.values(greyOptions)),
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
    hasUnderline: PropTypes.bool,
    hasHover: PropTypes.bool,
    hasUppercase: PropTypes.bool,
    transition: PropTypes.string,
};

Link.defaultProps = {
    colorType: colorTypeDefault,
    colorPallet: colorPalletDefault,
    colorTheme: colorThemeDefault,
    colorWab: greyDefault,
    colorStatus: formStatusDefault,
    hasUnderline: true,
    hasHover: true,
    hasUppercase: false,
    transition: null,
};

export default Link;
