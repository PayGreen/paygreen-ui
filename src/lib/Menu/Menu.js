import React from 'react';
import PropTypes from 'prop-types';
import { itemHtmlTagOptions, itemHtmlTagDefault } from '../../shared/constants';
import { MenuBase } from './style';

const Menu = props => {
    return (
        <MenuBase as={props.htmlTag} {...props}>
            {props.children}
        </MenuBase>
    );
};

Menu.propTypes = {
    htmlTag: PropTypes.oneOf(Object.values(itemHtmlTagOptions)),
};

Menu.defaultProps = {
    htmlTag: itemHtmlTagDefault,
};

export default Menu;
