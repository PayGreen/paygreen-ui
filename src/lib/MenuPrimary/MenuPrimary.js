import React from 'react';
import PropTypes from 'prop-types';
import { MenuPrimaryBase } from './style';

const MenuPrimary = props => {
    return <MenuPrimaryBase {...props}>{props.children}</MenuPrimaryBase>;
};

MenuPrimary.propTypes = {
    isOpen: PropTypes.bool,
};

MenuPrimary.defaultProps = {
    isOpen: PropTypes.false,
};

export default MenuPrimary;
