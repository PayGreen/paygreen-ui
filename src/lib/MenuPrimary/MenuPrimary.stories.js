import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean } from '@storybook/addon-knobs';
import {
    folder,
    colorPalletOptions,
    iconSizeOptions,
    spaceOptions,
} from '../../shared/constants';
import { LeafIcon, CardsIcon, OrganizationIcon, OutIcon } from '../Icon/Icon';
import Link from '../Link/Link';
import IconLabel from '../IconLabel/IconLabel';
import Text from '../Text/Text';
import MenuItem from '../MenuItem/MenuItem';
import MenuList from '../MenuList/MenuList';
import Menu from '../Menu/Menu';
import MenuPrimary from './MenuPrimary';

const subLink = {
    hasUnderline: false,
    hasHover: false,
};

storiesOf(folder.nav + 'MenuPrimary', module)
    .addDecorator(withKnobs)
    .add('MenuPrimary', () => (
        <MenuPrimary isOpen={boolean('Is open', true)}>
            <Menu>
                <a href="#">
                    <MenuItem isMain={true} hasHoverEmphasis={true}>
                        <Link
                            hasUnderline={false}
                            hasHover={false}
                            hasUppercase={true}
                        >
                            About us
                        </Link>
                    </MenuItem>
                </a>
            </Menu>

            <Menu>
                <MenuItem
                    isMain={true}
                    hasHoverBase={false}
                    hasHoverAlternative={false}
                >
                    <Link
                        hasUnderline={false}
                        hasHover={false}
                        colorPallet={colorPalletOptions.wab}
                        hasUppercase={true}
                    >
                        Our products
                    </Link>
                </MenuItem>

                <MenuList>
                    <a href="#">
                        <MenuItem hasHoverEmphasis={true}>
                            <LeafIcon iconSize={iconSizeOptions.lg} />

                            <Link {...subLink}>Tree</Link>

                            <Text>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit. Maecenas sit amet accumsan
                                dolor. Nullam fringilla quam leo.
                            </Text>
                        </MenuItem>
                    </a>

                    <a href="#">
                        <MenuItem hasHoverEmphasis={true}>
                            <CardsIcon iconSize={iconSizeOptions.lg} />

                            <Link {...subLink}>Payment</Link>

                            <IconLabel>
                                <OutIcon
                                    iconSize={iconSizeOptions.xs}
                                    marginRight={spaceOptions.xs}
                                />
                                Dev
                            </IconLabel>
                        </MenuItem>
                    </a>

                    <a href="#">
                        <MenuItem hasHoverEmphasis={true}>
                            <OrganizationIcon iconSize={iconSizeOptions.lg} />

                            <Link {...subLink}>Rounding</Link>
                        </MenuItem>
                    </a>
                </MenuList>
            </Menu>

            <Menu>
                <MenuItem
                    isMain={true}
                    hasHoverBase={false}
                    hasHoverAlternative={false}
                >
                    <Link
                        hasUnderline={false}
                        hasHover={false}
                        colorPallet={colorPalletOptions.wab}
                        hasUppercase={true}
                    >
                        Other products
                    </Link>
                </MenuItem>

                <MenuList>
                    <a href="#">
                        <MenuItem hasHoverEmphasis={true}>
                            <LeafIcon iconSize={iconSizeOptions.lg} />

                            <Link {...subLink}>Tree</Link>

                            <Text>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit. Maecenas sit amet accumsan
                                dolor. Nullam fringilla quam leo.
                            </Text>
                        </MenuItem>
                    </a>

                    <a href="#">
                        <MenuItem hasHoverEmphasis={true}>
                            <CardsIcon iconSize={iconSizeOptions.lg} />

                            <Link {...subLink}>Payment</Link>

                            <IconLabel>
                                <OutIcon
                                    iconSize={iconSizeOptions.xs}
                                    marginRight={spaceOptions.xs}
                                />
                                Dev
                            </IconLabel>
                        </MenuItem>
                    </a>

                    <a href="#">
                        <MenuItem hasHoverEmphasis={true}>
                            <OrganizationIcon iconSize={iconSizeOptions.lg} />

                            <Link {...subLink}>Rounding</Link>
                        </MenuItem>
                    </a>
                </MenuList>
            </Menu>
        </MenuPrimary>
    ));
