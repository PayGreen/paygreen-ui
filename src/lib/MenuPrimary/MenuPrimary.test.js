import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import {
    colorPalletOptions,
    iconSizeOptions,
    spaceOptions,
} from '../../shared/constants';
import { LeafIcon, CardsIcon, OutIcon } from '../Icon/Icon';
import Link from '../Link/Link';
import IconLabel from '../IconLabel/IconLabel';
import Text from '../Text/Text';
import MenuItem from '../MenuItem/MenuItem';
import MenuList from '../MenuList/MenuList';
import Menu from '../Menu/Menu';
import MenuPrimary from './MenuPrimary';

it('renders without crashing', () => {
    const menuPrimary = TestRenderer.create(
        <MenuPrimary theme={ThemeDefault} isOpen={true}>
            <Menu theme={ThemeDefault}>
                <a href="#">
                    <MenuItem isMain={true} theme={ThemeDefault}>
                        <Link
                            colorPallet={colorPalletOptions.wab}
                            hasUppercase={true}
                            theme={ThemeDefault}
                        >
                            About us
                        </Link>
                    </MenuItem>
                </a>
            </Menu>

            <Menu theme={ThemeDefault}>
                <a href="#">
                    <MenuItem
                        isMain={true}
                        hasHoverBase={false}
                        hasHoverAlternative={false}
                        theme={ThemeDefault}
                    >
                        <Link
                            colorPallet={colorPalletOptions.wab}
                            hasUppercase={true}
                            theme={ThemeDefault}
                        >
                            Other products
                        </Link>
                    </MenuItem>
                </a>

                <MenuList theme={ThemeDefault}>
                    <a href="#">
                        <MenuItem hasHoverEmphasis={true} theme={ThemeDefault}>
                            <LeafIcon
                                iconSize={iconSizeOptions.lg}
                                theme={ThemeDefault}
                            />

                            <Link theme={ThemeDefault}>Tree</Link>

                            <Text theme={ThemeDefault}>
                                Lorem ipsum dolor sit amet, consectetur
                                adipiscing elit. Maecenas sit amet accumsan
                                dolor. Nullam fringilla quam leo.
                            </Text>
                        </MenuItem>
                    </a>

                    <a href="#">
                        <MenuItem hasHoverEmphasis={true} theme={ThemeDefault}>
                            <CardsIcon
                                iconSize={iconSizeOptions.lg}
                                theme={ThemeDefault}
                            />

                            <Link theme={ThemeDefault}>Payment</Link>

                            <IconLabel theme={ThemeDefault}>
                                <OutIcon
                                    iconSize={iconSizeOptions.xs}
                                    marginRight={spaceOptions.xs}
                                    theme={ThemeDefault}
                                />
                                Dev
                            </IconLabel>
                        </MenuItem>
                    </a>
                </MenuList>
            </Menu>
        </MenuPrimary>,
    );
    expect(menuPrimary.toJSON()).toMatchSnapshot();
});
