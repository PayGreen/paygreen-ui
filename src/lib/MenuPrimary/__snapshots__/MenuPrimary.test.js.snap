// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul
  className="style__MenuPrimaryBase-sc-4ap9m-0 dqEskV"
>
  <li
    className="style__MenuBase-sc-6r1iet-0 EOKtT"
  >
    <a
      href="#"
    >
      <div
        className="style__MenuItemBase-sc-4wfueo-0 kQDKBR"
      >
        About us
      </div>
    </a>
  </li>
  <li
    className="style__MenuBase-sc-6r1iet-0 EOKtT"
  >
    <div
      className="style__MenuItemBase-sc-4wfueo-0 liCkWZ"
    >
      Other products
    </div>
    <ul
      className="style__MenuListBase-sc-144u7yf-0 kAyikQ"
    >
      <li>
        <a
          href="#"
        >
          <div
            className="style__MenuListItemBase-sc-81dvyg-0 fEBEyG"
          >
            <span
              className="style__IconBase-ak3679-0 weCXC icon"
            >
              <svg
                viewBox="0 0 143 143"
              >
                <path
                  d="M119.2 24.9c-.2-1.6-1.6-2.9-3.3-3-30.4-1.5-53.2 5.8-67.9 21.8C28.4 65 30 94.5 30.7 101.1c-4.7 8.6-9.5 17.7-9.5 17.7-.5 1.6.1 3.4 1.6 4.2.5.3 1.1.4 1.7.4 1.1 0 2.1-.5 2.8-1.4 0 0 .1-.1.1-.3.1-.1.1-.2.2-.2 0 0 2.9-5.4 3-5.5 1.6-2.9 3.5-6.5 5.5-10.2 22.6 4.1 47.4-2.8 63.8-18.1C111.3 77 124 57.4 119.2 24.9zm-66 23.5C64.3 36.3 81 29.7 102.9 28.8 95 33.8 83.7 41.5 72 51.6c.1-3.5 0-7.3-.3-9.8-.3-1.9-2-3.2-4-3-1.9.3-3.2 2-3 4 .5 3.4.3 11.1.1 15.2-9.7 9-19.3 19.7-27.3 31.7.5-11.1 3.6-28.1 15.7-41.3zM95 82.6c-14.2 13.3-35.5 19.6-55.3 16.7 4.7-7.9 10.2-15.2 16.1-21.9 3.3-.4 16.1-1.8 25.3.1 1.9.4 3.7-.8 4.1-2.7.4-1.9-.8-3.7-2.7-4.1-6.3-1.3-13.8-1.2-19.5-.9 7.2-7.3 14.6-13.8 21.5-19.3 3.2-.2 12.8-.6 18.6 0 1.9.2 3.7-1.1 3.9-3.1.2-1.9-1.1-3.7-3.1-3.9-2.8-.3-6.3-.4-9.7-.4 7.9-5.6 14.6-9.8 18.7-12.2 2.4 26.5-8.2 42.7-17.9 51.7z"
                />
              </svg>
            </span>
            <span
              className="style__LinkBase-sc-1gvcrtp-0 fjbDy"
            >
              Tree
            </span>
            <p
              className="style__TextBase-sc-11ixwvd-0 dwEpwZ"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sit amet accumsan dolor. Nullam fringilla quam leo.
            </p>
          </div>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <div
            className="style__MenuListItemBase-sc-81dvyg-0 fEBEyG"
          >
            <span
              className="style__IconBase-ak3679-0 weCXC icon"
            >
              <svg
                viewBox="0 0 143 143"
              >
                <path
                  d="M103.6 117.9H21.9c-5.6 0-10.2-4.6-10.2-10.2V61c0-5.6 4.6-10.2 10.2-10.2h81.8c5.6 0 10.2 4.6 10.2 10.2v46.8c.1 2.6-1 5.2-2.9 7.1-2 1.9-4.6 3-7.4 3zM21.9 57.8c-1.8 0-3.2 1.4-3.2 3.2v46.8c0 1.8 1.4 3.2 3.2 3.2h81.8c.9 0 1.7-.4 2.3-1 .6-.6.9-1.3.9-2.1V61.1c0-1.8-1.4-3.2-3.2-3.2l-81.8-.1z"
                />
                <path
                  d="M32 107.2c-.9 0-1.8-.3-2.5-1l-6.7-6.7c-.7-.7-1-1.5-1-2.5s.4-1.8 1-2.5l6.7-6.7c1.4-1.4 3.6-1.4 5 0s1.4 3.6 0 4.9L30.2 97l4.2 4.2c1.4 1.4 1.4 3.6 0 4.9-.7.7-1.6 1.1-2.4 1.1zM94.7 85.5H78.8c-4.4 0-8-3.6-8-8v-7.8c0-4.4 3.6-8 8-8h15.9c4.4 0 8 3.6 8 8v7.8c-.1 4.4-3.6 8-8 8zM78.8 68.8c-.5 0-1 .5-1 1v7.8c0 .5.5 1 1 1h15.9c.5 0 1-.5 1-1v-7.8c0-.5-.5-1-1-1H78.8z"
                />
                <path
                  d="M120.7 96c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5c1.8 0 3.2-1.4 3.2-3.2V39.1c0-1.8-1.4-3.2-3.2-3.2H38.9c-1.8 0-3.2 1.4-3.2 3.2v5.8c0 1.9-1.6 3.5-3.5 3.5s-3.5-1.6-3.5-3.5v-5.8c0-5.6 4.6-10.2 10.2-10.2h81.8c5.6 0 10.2 4.6 10.2 10.2v46.8c0 5.6-4.6 10.1-10.2 10.1z"
                />
              </svg>
            </span>
            <span
              className="style__LinkBase-sc-1gvcrtp-0 fjbDy"
            >
              Payment
            </span>
            <span
              className="style__IconLabelBase-sc-113x6jx-0 fZXecA"
            >
              <span
                className="style__IconBase-ak3679-0 evqAiC icon"
              >
                <svg
                  viewBox="0 0 143 143"
                >
                  <path
                    d="M116.3 76.9c-1.9 0-3.5 1.6-3.5 3.5v30.4c0 3-1.9 5.6-4 5.6H31.5c-2.1 0-4-2.6-4-5.6V36.7c0-3 1.8-5.6 4-5.6h38.6c1.9 0 3.5-1.6 3.5-3.5s-1.6-3.5-3.5-3.5H31.5c-6.1 0-11 5.6-11 12.6v74.1c0 6.9 4.9 12.6 11 12.6h77.3c6.1 0 11-5.6 11-12.6V80.4c0-2-1.6-3.5-3.5-3.5z"
                  />
                  <path
                    d="M125.6 42.1L94.7 18.7c-1.5-1.2-3.7-.9-4.9.7-1.2 1.5-.9 3.7.7 4.9l22.7 17.2c-26.6.4-42.2 3.9-51.8 11.2C49.9 61.6 47.5 75.6 47.5 94c0 1.9 1.6 3.5 3.5 3.5s3.5-1.6 3.5-3.5c0-33.3 8.2-44.6 58.3-45.5L90.5 65.1c-1.6 1.2-1.9 3.3-.7 4.9.7.9 1.7 1.4 2.8 1.4.7 0 1.5-.2 2.1-.7l30.9-23c.9-.7 1.4-1.7 1.4-2.8s-.5-2.1-1.4-2.8z"
                  />
                </svg>
              </span>
              Dev
            </span>
          </div>
        </a>
      </li>
    </ul>
  </li>
</ul>
`;
